kratos/python_scripts/print_info_in_file_process.py

```diff
@@ -0,0 +1,118 @@
+
+def Factory(settings, Model):
+    if not isinstance(settings, KratosMultiphysics.Parameters):
+        raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+    return PrintInfoInFileProcess(Model, settings["Parameters"])
+
+## All the processes python should be derived from "Process"
+class PrintInfoInFileProcess(KratosMultiphysics.OutputProcess):
+    """This process prints a text file with the required nodal or elemental information
+
+    Only the member variables listed below should be accessed directly.
+
+    Public member variables:
+    Model -- the container of the different model parts.
+    settings -- Kratos parameters containing solver settings.
+    """
+
+    def __init__(self, Model, settings ):
+        """ The default constructor of the class
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        Model -- the container of the different model parts.
+        settings -- Kratos parameters containing solver settings.
+        """
+        super().__init__()
+
+        default_settings = KratosMultiphysics.Parameters("""
+        {
+            "help"                     : "This process prints nodal/elemental information ina .txt file",
+            "mesh_id"                  : 0,
+            "model_part_name"          : "please_specify_model_part_name",
+            "variable_name"            : "SPECIFY_VARIABLE_NAME",
+            "variable_type"            : "nodal",
+            "file_name"                : "info_file.txt",
+            "output_control_type"      : "step",
+            "integration_point_number" : 0,
+            "erase_previous_info"      : true,
+            "output_interval"          : 1
+        }
+        """
+        )
+        settings.ValidateAndAssignDefaults(default_settings)
+
+        self.variable = KratosMultiphysics.KratosGlobals.GetVariable(settings["variable_name"].GetString())
+        self.is_nodal_variable_type = settings["variable_type"].GetString() == "nodal"
+        self.file_name = settings["file_name"].GetString()
+        self.output_control_type = settings["output_control_type"].GetString()
+        self.output_interval = settings["output_interval"].GetDouble()
+        self.model_part = Model[settings["model_part_name"].GetString()]
+        self.instant_previous_plot = 0.0
+        self.integration_point = settings["integration_point_number"].GetInt()
+
+        open_file_aproach = "a"
+        if settings["erase_previous_info"].GetBool():
+            open_file_aproach = "w"
+
+        self.plot_file = open(self.file_name, open_file_aproach)
+        self.plot_file.write("In this file we print the " + settings["variable_type"].GetString() + " " + settings["variable_name"].GetString() + " in the sub_model_part: " + settings["model_part_name"].GetString() + "\n\n")
+        self.plot_file.write(" TIME      \t" + settings["variable_name"].GetString() + "\n")
+        self.plot_file.close()
```

Comment:
please use `with open(...)`
alternatively you could use the file logger

Reply:
Do you propose to use the logger?

Reply:
thinking twice this should actually the [time_based_ascii_file_writer_utility](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/python_scripts/time_based_ascii_file_writer_utility.py)  so that it works properly for restarts